app/src/main/java/com/example/una/LoginActivity.java

```diff
@@ -0,0 +1,109 @@
+            new AuthUI.IdpConfig.EmailBuilder().build(),
+            new AuthUI.IdpConfig.GoogleBuilder().build(),
+            new AuthUI.IdpConfig.FacebookBuilder().build());
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+        context = this;
+
+        auth = FirebaseAuth.getInstance();
+    }
+
+    // start the login process
+    public void updateUI(FirebaseUser user) {
+        if (user != null) {
+            Log.i(TAG, user.getEmail() + " is already signed in -- updatingUI");
+            startHomeActivity(user);
+        } else {
+            // Choose authentication providers
+            List<AuthUI.IdpConfig> providers = Arrays.asList(
+                    new AuthUI.IdpConfig.EmailBuilder().build(),
+                    new AuthUI.IdpConfig.GoogleBuilder().build(),
+                    new AuthUI.IdpConfig.FacebookBuilder().build());
+
+            // Create and launch sign-in intent
+            startActivityForResult(
+                    AuthUI.getInstance()
+                            .createSignInIntentBuilder()
+                            .setAvailableProviders(providers)
+                            .setLogo(R.drawable.ic_cuppa_24dp)
+                            .setTosAndPrivacyPolicyUrls(
+                                    "https://una.co/terms.html",
+                                    "https://una.co/privacy.html")
+                            .build(),
+                    RC_SIGN_IN);
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == RC_SIGN_IN) {
+            IdpResponse response = IdpResponse.fromResultIntent(data);
+            if (resultCode == RESULT_OK) {
+                Log.i(TAG, "starting home activity from activity result");
+                // user is signed in
+                FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+                startHomeActivity(user);
+            } else {
+                // Sign in failed
+                if (response == null) {
+                    Toast.makeText(this, "Sign in Cancelled", Toast.LENGTH_SHORT).show();
+                    return;
+                } else if (response.getError().getErrorCode() == ErrorCodes.NO_NETWORK) {
+                    Toast.makeText(this, "No Internet Connection", Toast.LENGTH_SHORT).show();
+                } else {
+                    Toast.makeText(this, "Unable to Complete Sign In", Toast.LENGTH_SHORT).show();
+                }
+            }
+        }
+    }
+
+    private void startHomeActivity(FirebaseUser user) {
+        Log.i(TAG, "starting home activity");
+        Intent homeIntent = new Intent(this, MainActivity.class);
+        homeIntent.putExtra("user", user);
+        startActivity(homeIntent);
```

Comment:
you will want to `finish()` this activity once the other activity is started. Otherwise users will come back to sign-in screen when they hit back on `MainActivity`